server/commands/battle.py

```diff
@@ -0,0 +1,597 @@
+        fighter_name = client.area.area_manager.char_list[client.char_id]
+        if client.battle.status == "poison":
+            client.battle.hp += -client.battle.maxhp / area.battle_poison_damage
+            area.send_ic(client=None, msg_type="1", pre=client.last_sprite, folder=fighter_name, anim=client.last_sprite, msg=f"~{client.battle.fighter}~ is affected by poisoning and loses {client.battle.maxhp / area.battle_poison_damage} hp", pos=client.pos, sfx="", emote_mod=0, cid=-1, sfx_delay=0, button=0, evidence=[0], flip=client.flip, ding=0, color=3, showname="", charid_pair=client.charid_pair, other_folder="", other_emote="", offset_pair=client.offset_pair, other_offset=0, other_flip=0, nonint_pre=0, sfx_looping="0", screenshake=0, frames_shake="", frames_realization="", frames_sfx="", additive=0, effect="poison", targets=None)
+        if client.battle.hp < 1:
+            area.fighters.remove(client)
+            client.battle = ClientManager.Battle_char(client, client.battle.fighter)
+            area.send_ic(client=None, msg_type="1", pre=client.last_sprite, folder=fighter_name, anim=client.last_sprite, msg=f"~{client.battle.fighter}~ dies...", pos=client.pos, sfx="", emote_mod=0, cid=-1, sfx_delay=0, button=0, evidence=[0], flip=client.flip, ding=0, color=3, showname="", charid_pair=client.charid_pair, other_folder="", other_emote="", offset_pair=100, other_offset=0, other_flip=0, nonint_pre=0, sfx_looping="0", screenshake=0, frames_shake="", frames_realization="", frames_sfx="", additive=0, effect="", targets=None)
+        send_info_fighter(client)
+        msg = send_battle_info(client)
+        client.send_ooc(msg)
+    if len(area.fighters) == 1:
+        winner = area.fighters[0]
+        fighter_name = winner.area.area_manager.char_list[winner.char_id]
+        area.send_ic(client=None, msg_type="1", pre=1, folder=fighter_name, anim=client.last_sprite, msg=f"~{client.battle.fighter}~ wins the battle!", pos=client.pos, sfx="", emote_mod=0, cid=-1, sfx_delay=0, button=0, evidence=[0], flip=client.flip, ding=0, color=3, showname="", charid_pair=client.charid_pair, other_folder="", other_emote="", offset_pair=0, other_offset=0, other_flip=0, nonint_pre=0, sfx_looping="0", screenshake=0, frames_shake="", frames_realization="", frames_sfx="", additive=0, effect="", targets=None)
+        winner.battle = ClientManager.Battle_char(winner, winner.battle.fighter)
+    if len(area.fighters) == 0:
+        area.send_ic(client=None, msg_type="1", pre=1, folder=fighter_name, anim=client.last_sprite, msg=f"~Everyone~ died...", pos=client.pos, sfx="", emote_mod=0, cid=-1, sfx_delay=0, button=0, evidence=[0], flip=client.flip, ding=0, color=3, showname="", charid_pair=client.charid_pair, other_folder="", other_emote="", offset_pair=0, other_offset=0, other_flip=0, nonint_pre=0, sfx_looping="0", screenshake=0, frames_shake="", frames_realization="", frames_sfx="", additive=0, effect="", targets=None)
+    return area.fighters
```

Comment:
oh god this is a nightmare. maybe just have 1 send_ic at the end that it uses to send the message??? check out how the Cross-Swords/Scrum Debate code does it

Reply:
Done!